pkg/volume/iscsi/iscsi_util.go

```diff
@@ -457,61 +460,32 @@ func (util *ISCSIUtil) AttachDisk(b iscsiDiskMounter) (string, error) {
 	}
 
 	klog.V(5).Infof("iscsi: AttachDisk devicePath: %s", devicePath)
-	// run global mount path related operations based on volumeMode
-	return globalPDPathOperation(b)(b, devicePath, util)
+
+	if err = util.persistISCSI(b); err != nil {
+		// Return transient error so kubelet calls Unmount / Unmap when the pod
+		// is deleted.
+		return "", types.NewTransientOperationFailure(err.Error())
```

Comment:
transient error will not cause volume to be recorded as mounted. We need `NewUncertainProgressError` for that.

Reply:
Fixed